src/components/ebay-snackbar-dialog/index.marko

```diff
@@ -0,0 +1,105 @@
+import processHtmlAttributes from "../../common/html-attributes"
+
+static var ignoredAttributes = [
+    "open"
+];
+
+class {
+    clearAllTimeouts() {
+        while(this.timeouts.length) {
+            clearTimeout(this.timeouts.pop());
+        }
+    }
+
+    onInput(input) {
+        this.clearAllTimeouts();
+        this.state = {open: input.open || this.state.open || false}
+        if (input.open !== false) {
+            this.timeouts.push(setTimeout(() => {
+                this.emit("close");
+                this.state.open = false;
+            }, input.timeout || this.DEFAULT_TIMEOUT_LENGTH));
+        }
+    }
+    onDestroy() {
+        this.clearAllTimeouts();
+    }
+
+    onCreate() {
+        this.DEFAULT_TIMEOUT_LENGTH = 6000;
```

Comment:
I would put this as a global variable outside the class scope (it works better in `component.js`)